qiskit/circuit/library/evolved_operator_ansatz.py

```diff
@@ -233,6 +239,15 @@ def _build(self):
         if self.initial_state:
             evolution.compose(self.initial_state, front=True, inplace=True)
 
+        # cast global phase to float if it has no free parameters
+        if isinstance(evolution.global_phase, ParameterExpression):
+            # no free parameters
+            if len(evolution.global_phase.parameters) == 0:
+                evolution.global_phase = float(evolution.global_phase)
+            # parameters sum up to 0 (but ParameterExpression still contains free parameters)
+            if evolution.global_phase._symbol_expr == 0:
+                evolution.global_phase = 0.0
```

Comment:
is this logically an `elif`?  I suppose they shouldn't cause different values, but it seems confusing.

Also, would it be appropriate to do
```python
try:
    evolution.global_phase = float(evolution.global_phase._symbol_expr)
except TypeError:
    pass
```
to allow cases where the global phase evaluates to any knowable float, not just zero?  I don't really know the context, so I don't know if that's something that could reasonably happen, though.

Reply:
great suggestions, I updated 👍🏻